### PR TITLE
NFT-1019: Track LP positions on subgraph

### PR DESCRIPTION
**Pull request by adamgobes**

### CODE REVIEW

File: src/position-manager.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  NonfungiblePositionManager,
+  Transfer,
+} from "../generated/NonfungiblePositionManager/NonfungiblePositionManager";
+import { ERC20Token, PaprController, Position } from "../generated/schema";
+import { handleSyntheticSwapFromLP } from "./activity";
+import { loadOrCreateERC20Token } from "./utils";
+
+const paprControllerAddress = "0x3b29c19ff2fcea0ff98d0ef5b184354d74ea74b0";
+
+function isPaprPool(
```
Comment (wilsoncusack) [inline src/position-manager.ts:15]: theoretically we'd want to track all papr pools, but we can save that complexity for another day

File: src/pool.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import { Address, dataSource } from "@graphprotocol/graph-ts";
```
Comment (adamgobes) [inline src/pool.ts:1]: this is swaps.ts renamed

File: schema.graphql [hunk 3]
```diff
@@ -171,13 +171,24 @@ type ActivityRemovedCollateral @entity {
   activity: Activity!
 }
 
+type Position @entity {
+  id: ID!
+  user: Bytes!
+  tickLower: Int!
+  tickUpper: Int!
+  liquidity: BigInt!
+  amount0: BigInt!
```
Comment (wilsoncusack) [inline schema.graphql:180]: maybe should label cumulative, so it's clear?

File: schema.graphql [hunk 4]
```diff
@@ -171,13 +171,24 @@ type ActivityRemovedCollateral @entity {
   activity: Activity!
 }
 
+type Position @entity {
+  id: ID!
+  user: Bytes!
+  tickLower: Int!
+  tickUpper: Int!
+  liquidity: BigInt!
```
Comment (wilsoncusack) [inline schema.graphql:179]: is this cumulative liquidity?

File: schema.graphql [hunk 5]
```diff
@@ -191,4 +202,13 @@ type Activity @entity {
   auctionCollateral: ERC721Token
   auctionTokenId: BigInt
   auctionEndPrice: BigInt
+  position: Position
```
Comment (wilsoncusack) [inline schema.graphql:205]: naming could be more clear, uniswapLPPosition or something

File: schema.graphql [hunk 6]
```diff
@@ -191,4 +202,13 @@ type Activity @entity {
...
+  totalLiquidityAdded: BigInt
+  totalLiquidity0: BigInt
```
Comment (wilsoncusack) [inline schema.graphql:207]: I think totalLiquidity0 is kinda confusing: this is cumulative token0 right? also I think we only need to store the cumulative, do we need to know how much was added here? Was thinking we could just look at the prev cumulative and know the diff

Comment (adamgobes) [reply schema.graphql:207]: yeah we need to know how much was added for the activity feed

File: schema.graphql [hunk 7]
```diff
@@ -191,4 +202,13 @@ type Activity @entity {
...
+  totalLiquidity1: BigInt
+  liquidityAdded: BigInt
+  liquidityAdded0: BigInt
+  liquidityAdded1: BigInt
+  positionTickUpper: Int
```
[the remark this answers is not in the document]
Comment (wilsoncusack) [reply schema.graphql:212]: agree

**Pull request merged**